From d5860b09f462d20f5881d58317c062cb06829f9c Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Fri, 8 Dec 2023 10:52:26 +0000
Subject: [PATCH] Q_ENUMS cannot be use signed long long

---
 src/qofonoextcell.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qofonoextcell.h b/src/qofonoextcell.h
index 87b632b..9aaf63e 100644
--- a/src/qofonoextcell.h
+++ b/src/qofonoextcell.h
@@ -66,7 +66,7 @@ public:
         UNKNOWN = Unknown // For backward compatibility
     };
 
-    enum Constants {
+    enum Constants : quint64 {
         InvalidValue = INT_MAX,
         InvalidValue64 = LLONG_MAX
     };
-- 
2.43.0

