From 4853011d8143a5b308e3334d563a232030cb7b16 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Fri, 11 Aug 2023 07:30:55 +0000
Subject: [PATCH 2/2] distinguish wakeup type in AlarmObject

---
 src/alarmobject.cpp | 8 ++++++--
 src/alarmobject.h   | 3 ++-
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/alarmobject.cpp b/src/alarmobject.cpp
index 284aa26..19fd186 100644
--- a/src/alarmobject.cpp
+++ b/src/alarmobject.cpp
@@ -201,14 +201,14 @@
 
 AlarmObject::AlarmObject(QObject *parent)
     : QObject(parent), m_hour(0), m_minute(0), m_second(0), m_enabled(false),
-      m_createdDate(QDateTime::currentDateTime()), m_countdown(false), m_reminder(false), m_triggerTime(0),
+      m_createdDate(QDateTime::currentDateTime()), m_countdown(false), m_reminder(false), m_wakeup(false), m_triggerTime(0),
       m_elapsed(0), m_cookie(0), m_timeoutSnoozeCounter(0), m_maximalTimeoutSnoozeCount(0)
 {
 }
 
 AlarmObject::AlarmObject(const QMap<QString,QString> &data, QObject *parent)
     : QObject(parent), m_hour(0), m_minute(0), m_second(0), m_enabled(false),
-      m_createdDate(QDateTime::currentDateTime()), m_countdown(false), m_reminder(false), m_triggerTime(0),
+      m_createdDate(QDateTime::currentDateTime()), m_countdown(false), m_reminder(false), m_wakeup(false), m_triggerTime(0),
       m_elapsed(0), m_cookie(0)
 {
     for (QMap<QString,QString>::ConstIterator it = data.begin(); it != data.end(); it++) {
@@ -261,6 +261,8 @@ AlarmObject::AlarmObject(const QMap<QString,QString> &data, QObject *parent)
             m_phoneNumber = it.value();
         } else if (it.key() == QLatin1String("type") && it.value() == QLatin1String("reminder")) {
             m_reminder = true;
+        } else if (it.key() == QLatin1String("type") && it.value() == QLatin1String("wakeup")) {
+            m_wakeup = true;
         }
     }
 
@@ -372,6 +374,8 @@ int AlarmObject::type() const
         return Calendar;
     else if (m_countdown)
         return Countdown;
+    else if (m_wakeup)
+        return WakeUp;
     else
         return Clock;
 }
diff --git a/src/alarmobject.h b/src/alarmobject.h
index ccf2fb2..f567a41 100644
--- a/src/alarmobject.h
+++ b/src/alarmobject.h
@@ -70,7 +70,7 @@ public:
     AlarmObject(QObject *parent = 0);
     AlarmObject(const QMap<QString,QString> &data, QObject *parent = 0);
 
-    enum Type { Calendar, Clock, Countdown, Reminder };
+    enum Type { Calendar, Clock, Countdown, Reminder, WakeUp };
     Q_ENUMS(Type)
 
     QString title() const { return m_title; }
@@ -161,6 +161,7 @@ protected:
     QDateTime m_createdDate;
     bool m_countdown;
     bool m_reminder;
+    bool m_wakeup;
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
     qint64 m_triggerTime;
     qint64 m_elapsed;
-- 
2.41.0

