From df750a3402682870327ae0ecee940eae834a07e5 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Sun, 16 Jul 2023 05:48:47 +0000
Subject: [PATCH] operator== const

---
 lib/ofononetworkoperator.cpp | 2 +-
 lib/ofononetworkoperator.h   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/ofononetworkoperator.cpp b/lib/ofononetworkoperator.cpp
index 64fb37b..5132b9e 100644
--- a/lib/ofononetworkoperator.cpp
+++ b/lib/ofononetworkoperator.cpp
@@ -45,7 +45,7 @@ OfonoNetworkOperator::OfonoNetworkOperator(const OfonoNetworkOperator& op)
             this, SLOT(propertyChanged(const QString&, const QVariant&)));
 }    
     
-bool OfonoNetworkOperator::operator==(const OfonoNetworkOperator &op)
+bool OfonoNetworkOperator::operator==(const OfonoNetworkOperator &op) const
 {
     return path() == op.path();
 }
diff --git a/lib/ofononetworkoperator.h b/lib/ofononetworkoperator.h
index 62c97af..7740559 100644
--- a/lib/ofononetworkoperator.h
+++ b/lib/ofononetworkoperator.h
@@ -60,7 +60,7 @@ public:
     ~OfonoNetworkOperator();
 
     OfonoNetworkOperator operator=(const OfonoNetworkOperator &op);
-    bool operator==(const OfonoNetworkOperator &op);
+    bool operator==(const OfonoNetworkOperator &op) const;
 
     //! Returns the D-Bus object path of the network operator object
     QString path() const;
-- 
2.41.0

