From de01c30498f2b78a304994801bb773cdfddc7373 Mon Sep 17 00:00:00 2001
From: Chupligin Sergey <neochapay@gmail.com>
Date: Mon, 24 Jul 2023 20:48:11 +0000
Subject: [PATCH] Fixup build on qt6

---
 src/plugin/plugin.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/plugin/plugin.cpp b/src/plugin/plugin.cpp
index aeb4600..e661d0b 100644
--- a/src/plugin/plugin.cpp
+++ b/src/plugin/plugin.cpp
@@ -75,7 +75,11 @@ public:
         qDBusRegisterMetaType<NemoDeviceLock::Fingerprint>();
         qDBusRegisterMetaType<QVector<NemoDeviceLock::Fingerprint>>();
 
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
+        qmlRegisterAnonymousType<NemoDeviceLock::FingerprintModel>(uri, 1);
+#else
         qmlRegisterType<NemoDeviceLock::FingerprintModel>();
+#endif
 
         qmlRegisterSingletonType<NemoDeviceLock::DeviceLock>(uri, 1, 0, "DeviceLock", createDeviceLock);
 
-- 
2.49.0

