diff --git a/Makefile.am b/Makefile.am
index dc70f42..d68a233 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -11,9 +11,6 @@ MAINTAINERCLEANFILES = \
         build-stamp compile depcomp acinclude.m4 aclocal.m4 \
 	stamp-h1 
 
-clean-local:
-	dh_clean
-
 $(top_srcdir)/.version:
 	echo $(VERSION) > $@-t && mv $@-t $@
 
diff --git a/configure.ac b/configure.ac
index 363b09b..8dee332 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,7 +73,7 @@ PKG_CHECK_MODULES(LIBPULSECORE, [pulsecore])
 AC_SUBST(LIBPULSECORE_CFLAGS)
 AC_SUBST(LIBPULSECORE_LIBS)
 
-PKG_CHECK_MODULES([MEEGOCOMMON], [libmeego-common >= 24])
+PKG_CHECK_MODULES([MEEGOCOMMON], [libmeego-common >= 15])
 AC_SUBST(MEEGOCOMMON_CFLAGS)
 AC_SUBST(MEEGOCOMMON_LIBS)
 
