From 51ff191a7c6c2319215fefc6162dcc0448bc859a Mon Sep 17 00:00:00 2001
From: Sergey Chupligin <neochapay@gmail.com>
Date: Fri, 25 Feb 2022 15:11:39 +0000
Subject: [PATCH] geoclue use /usr/lib

---
 src/locationsettings.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/locationsettings.cpp b/src/locationsettings.cpp
index 2cea146..100ed29 100644
--- a/src/locationsettings.cpp
+++ b/src/locationsettings.cpp
@@ -246,20 +246,20 @@ void LocationSettingsPrivate::loadProviders()
 {
     // for now just hard-coding the known potential providers.
     // can be replaced with config type of thing if there's need to support more and more providers.
-    if (QFile::exists(QStringLiteral("/usr/libexec/geoclue-here"))) {
+    if (QFile::exists(QStringLiteral("/usr/lib/geoclue-here"))) {
         LocationProvider provider;
         provider.hasAgreement = true;
         m_providers[HereName] = provider;
     }
 
-    if (QFile::exists(QStringLiteral("/usr/libexec/geoclue-mlsdb"))) {
+    if (QFile::exists(QStringLiteral("/usr/lib/geoclue-mlsdb"))) {
         LocationProvider provider;
         provider.hasAgreement = true;
         provider.offlineCapable = true;
         m_providers[MlsName] = provider;
     }
 
-    if (QFile::exists(QStringLiteral("/usr/libexec/geoclue-yandex"))) {
+    if (QFile::exists(QStringLiteral("/usr/lib/geoclue-yandex"))) {
         LocationProvider provider;
         provider.hasAgreement = true; // supposedly
         m_providers[YandexName] = provider;
-- 
2.35.1

