diff --git a/src/src.pro b/src/src.pro
index 9929a1b..ac8941d 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -101,6 +101,6 @@ QMAKE_PKGCONFIG_DESCRIPTION = System settings application development files
 QMAKE_PKGCONFIG_LIBDIR = $$target.path
 QMAKE_PKGCONFIG_INCDIR = $$develheaders.path
 QMAKE_PKGCONFIG_DESTDIR = pkgconfig
-QMAKE_PKGCONFIG_REQUIRES = Qt5Core Qt5DBus profile libsailfishkeyprovider connman-qt5 nemodbus
+QMAKE_PKGCONFIG_REQUIRES = Qt5Core Qt5DBus profile connman-qt5 nemodbus
 
 INSTALLS += target develheaders pkgconfig locationconfig compat_locationconfig
