From d3b7fabdc65231102f32a65191d12ba032851691 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Fri, 22 Apr 2022 13:39:32 +0200
Subject: [PATCH] fix supported-lanauges directory

---
 src/languagemodel.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/languagemodel.cpp b/src/languagemodel.cpp
index 1de1718..c1a017d 100644
--- a/src/languagemodel.cpp
+++ b/src/languagemodel.cpp
@@ -42,7 +42,7 @@
 #include <nemo-dbus/interface.h>
 
 namespace {
-const char * const LanguageSupportDirectory = "/usr/share/jolla-supported-languages";
+const char * const LanguageSupportDirectory = "/usr/share/supported-languages";
 
 bool nameLessThan(const Language &lang1, const Language &lang2)
 {
-- 
2.35.1

