diff --git a/src/src.pro b/src/src.pro
index 2c90219..e89330b 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -1,5 +1,5 @@
 TEMPLATE = lib
-TARGET = usb-moded-qt5
+TARGET = usb-moded-qt$${QT_MAJOR_VERSION}
 CONFIG += link_pkgconfig
 PKGCONFIG += usb_moded
 
@@ -13,6 +13,8 @@ CONFIG(debug, debug|release) {
   DEFINES += QUSBMODED_DEBUG=1
 }
 
+PREFIX = /usr
+
 SOURCES += \
     qusbmode.cpp \
     qusbmoded.cpp
@@ -32,7 +34,7 @@ com_meego_usb_moded.files = $$USB_MODED_INCLUDE_PATH/com.meego.usb_moded.xml
 com_meego_usb_moded.header_flags = -N -c QUsbModedInterface
 com_meego_usb_moded.source_flags = -N -c QUsbModedInterface
 
-headers.path = $$INSTALL_ROOT$$PREFIX/include/usb-moded-qt5
+headers.path = $$INSTALL_ROOT$$PREFIX/include/usb-moded-qt$${QT_MAJOR_VERSION}
 headers.files = $$PUBLIC_HEADERS
 INSTALLS += headers
 
@@ -40,7 +42,7 @@ target.path = $$[QT_INSTALL_LIBS]
 INSTALLS += target
 
 pkgconfig.files = $$TARGET.pc
-pkgconfig.path = $$target.path/pkgconfig-qt5
+pkgconfig.path = $$target.path/pkgconfig-qt$${QT_MAJOR_VERSION}
 QMAKE_PKGCONFIG_NAME = lib$$TARGET
 QMAKE_PKGCONFIG_VERSION = $$VERSION
 QMAKE_PKGCONFIG_DESCRIPTION = Qt bindings for usb_moded
