From 59cf5916edc682d67b0d033a1e711b5deaa50834 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Fri, 8 Dec 2023 09:01:30 +0000
Subject: [PATCH] trouble with c++11

---
 features/qt_common.prf | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/features/qt_common.prf b/features/qt_common.prf
index f51b326..612d9b2 100644
--- a/features/qt_common.prf
+++ b/features/qt_common.prf
@@ -14,12 +14,6 @@ CONFIG -= debug_and_release_target
 
 DEFINES *= QT_NO_NARROWING_CONVERSIONS_IN_CONNECT
 
-qtConfig(c++11): CONFIG += c++11 strict_c++
-qtConfig(c++14): CONFIG += c++14
-qtConfig(c++1z): CONFIG += c++1z
-qtConfig(c++2a): CONFIG += c++2a
-qtConfig(c99): CONFIG += c99
-qtConfig(c11): CONFIG += c11
 qtConfig(separate_debug_info): CONFIG += separate_debug_info
 qtConfig(stack-protector-strong): CONFIG += stack_protector_strong
 contains(TEMPLATE, .*lib) {
-- 
2.43.0

