diff --git a/features/qt_common.prf b/features/qt_common.prf
index 612d9b2..3e97710 100644
--- a/features/qt_common.prf
+++ b/features/qt_common.prf
@@ -15,7 +15,6 @@ CONFIG -= debug_and_release_target
 DEFINES *= QT_NO_NARROWING_CONVERSIONS_IN_CONNECT
 
 qtConfig(separate_debug_info): CONFIG += separate_debug_info
-qtConfig(stack-protector-strong): CONFIG += stack_protector_strong
 contains(TEMPLATE, .*lib) {
     # module and plugins
     unix:qtConfig(reduce_relocations): CONFIG += bsymbolic_functions
