From acc6d5f293321507462ccca418a10be2269c2884 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Sun, 27 Apr 2025 07:34:08 +0000
Subject: [PATCH] Compatibility with CMake < 3.10 will be removed from a future
 version of CMake

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fceda1c..f52e6b4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.10)
 
 option(NATIVE_VIBRATOR "Use vibra directly through kernel interface instead of using HAL" OFF)
 
-- 
2.49.0

