From df679c17381062b0e7d35bbd5883072efd844baf Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Sun, 9 Jul 2023 09:42:15 +0000
Subject: [PATCH] Fix use with Qt6

---
 lib/lib.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/lib.pro b/lib/lib.pro
index e50f80f..e129e49 100644
--- a/lib/lib.pro
+++ b/lib/lib.pro
@@ -49,6 +49,6 @@ QMAKE_PKGCONFIG_DESCRIPTION = Nemomobile CPU/display keepalive development files
 QMAKE_PKGCONFIG_LIBDIR      = $$target.path
 QMAKE_PKGCONFIG_INCDIR      = $$develheaders.path
 QMAKE_PKGCONFIG_DESTDIR     = pkgconfig
-QMAKE_PKGCONFIG_REQUIRES    = Qt5Core Qt5DBus
+QMAKE_PKGCONFIG_REQUIRES    = Qt$${QT_MAJOR_VERSION}Core Qt$${QT_MAJOR_VERSION}DBus
 
 INSTALLS += target develheaders pkgconfig
-- 
2.40.1

