diff --git a/plugin/mlsdbonlinelocator.cpp b/plugin/mlsdbonlinelocator.cpp
index 0759662..a682909 100644
--- a/plugin/mlsdbonlinelocator.cpp
+++ b/plugin/mlsdbonlinelocator.cpp
@@ -214,11 +214,6 @@ QPair<QDateTime, QVariantMap> MlsdbOnlineLocator::buildLocationQuery(
 
 bool MlsdbOnlineLocator::findLocation(const QPair<QDateTime, QVariantMap> &query)
 {
-    if (query.first.isNull() || query.second.isEmpty()) {
-        qCDebug(lcGeoclueMlsdbOnline) << "Empty query data provided";
-        return false;
-    }
-
     if (m_currentReply) {
         qCDebug(lcGeoclueMlsdbOnline) << "Previous request still in progress";
         return true;
diff --git a/plugin/mlsdbprovider.cpp b/plugin/mlsdbprovider.cpp
index 1cbd60b..d1a1dd8 100644
--- a/plugin/mlsdbprovider.cpp
+++ b/plugin/mlsdbprovider.cpp
@@ -40,8 +40,8 @@ namespace {
     const quint32 MinimumInterval = 10000;      // 10s, the shortest interval at which the plugin will recalculate position since last update
     const quint32 ReuseInterval = 30000;        // 30s, the amount of time a previously calculated position updates will be re-used for without recalculating new position
     const quint32 FallbackInterval = 120000;    // 120s, the amount of time a previously calculated position update with high accuracy can supercede a newly calculated low-accuracy position
-    const QString LocationSettingsDir = QStringLiteral("/var/lib/location/");
-    const QString LocationSettingsFile = QStringLiteral("/var/lib/location/location.conf");
+    const QString LocationSettingsDir = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/location/";
+    const QString LocationSettingsFile = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation) + "/location/location.conf";
     const QString LocationSettingsEnabledKey = QStringLiteral("location/enabled");
     const QString LocationSettingsMlsEnabledKey = QStringLiteral("location/mls/enabled");
     const QString LocationSettingsMlsOnlineEnabledKey = QStringLiteral("location/mls/online_enabled");
