From ae42128b1eb8b501c66133002150e7c73f18f336 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Sat, 12 Jun 2021 12:44:44 +0200
Subject: [PATCH] avoid warning "Compatibility with CMake < 2.8.12 will be
 removed"

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3392d6e..ed52b34 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 PROJECT(cor)
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.20)
 
 find_package(PkgConfig)
 find_package (Threads)
-- 
2.31.1

