diff --git a/declarative/src/plugin.cpp b/declarative/src/plugin.cpp
index 0aac399..02c66b7 100644
--- a/declarative/src/plugin.cpp
+++ b/declarative/src/plugin.cpp
@@ -73,6 +73,6 @@ void CommHistoryPlugin::registerTypes(const char *uri)
     qmlRegisterType<DraftEvent>(uri, 1, 0, "DraftEvent");
     qmlRegisterType<MmsHelper>(uri, 1, 0, "MmsHelper");
 
-    qmlRegisterType<CommHistory::GroupObject>();
-    qmlRegisterType<CommHistory::ContactGroup>();
+    qmlRegisterAnonymousType<CommHistory::GroupObject>(uri, 0);
+    qmlRegisterAnonymousType<CommHistory::ContactGroup>(uri, 0);
 }
diff --git a/libcommhistory.pro b/libcommhistory.pro
index f23e67f..7dcfff5 100644
--- a/libcommhistory.pro
+++ b/libcommhistory.pro
@@ -25,12 +25,10 @@
 TEMPLATE  = subdirs
 SUBDIRS   = src   \
             declarative \
-            tools \
-            tests
+            tools
 
 declarative.depends = src
 tools.depends = src
-tests.depends = src
 
 OTHER_FILES += rpm/libcommhistory-qt5.spec
 
diff --git a/src/contactfetcher.cpp b/src/contactfetcher.cpp
index b0c5e38..b4b71fb 100644
--- a/src/contactfetcher.cpp
+++ b/src/contactfetcher.cpp
@@ -67,7 +67,7 @@ ContactFetcherPrivate::ContactFetcherPrivate(ContactFetcher *parent)
     , m_resolver(0)
     , m_fetching(false)
 {
-    SeasideCache::registerChangeListener(this, SeasideCache::FetchAvatar);
+    SeasideCache::registerChangeListener(this);
 }
 
 ContactFetcherPrivate::~ContactFetcherPrivate()
diff --git a/src/contactgroupmodel.cpp b/src/contactgroupmodel.cpp
index 31f713e..942fc40 100644
--- a/src/contactgroupmodel.cpp
+++ b/src/contactgroupmodel.cpp
@@ -144,7 +144,7 @@ void ContactGroupModelPrivate::setManager(GroupManager *m)
             emit q->contactGroupCreated(items[index]);
         }
 
-        qSort(items.begin(), items.end(), contactGroupSort);
+        std::sort(items.begin(), items.end(), contactGroupSort);
     }
 
     q->endResetModel();
diff --git a/src/contactlistener.cpp b/src/contactlistener.cpp
index 69bad1a..e0c9d8c 100644
--- a/src/contactlistener.cpp
+++ b/src/contactlistener.cpp
@@ -31,6 +31,7 @@
 #include <QContactOnlineAccount>
 #include <QContactPhoneNumber>
 #include <QContactEmailAddress>
+#include <QContactSyncTarget>
 
 #include "commonutils.h"
 #include "contactresolver.h"
@@ -100,7 +101,7 @@ ContactListenerPrivate::ContactListenerPrivate(ContactListener *q)
     , retryResolver(0)
     , q_ptr(q)
 {
-    SeasideCache::registerChangeListener(this, SeasideCache::FetchAvatar);
+    SeasideCache::registerChangeListener(this);
 }
 
 ContactListenerPrivate::~ContactListenerPrivate()
@@ -162,7 +163,9 @@ void ContactListenerPrivate::itemUpdated(SeasideCache::CacheItem *item)
     Q_Q(ContactListener);
 
     // Only aggregate contacts are relevant
-    if (item->contact.collectionId() != SeasideCache::aggregateCollectionId())
+    static const QString aggregateTarget(QString::fromLatin1("aggregate"));
+    QContactSyncTarget syncTarget(item->contact.detail<QContactSyncTarget>());
+    if (syncTarget.syncTarget() != aggregateTarget)
         return;
 
     // Make a list of Recipient from the contacts addresses to compare against
diff --git a/src/conversationmodel.cpp b/src/conversationmodel.cpp
index 642554b..6e1fe44 100644
--- a/src/conversationmodel.cpp
+++ b/src/conversationmodel.cpp
@@ -71,7 +71,7 @@ void ConversationModelPrivate::groupsAddedSlot(const QList<Group> &/*groups*/)
         if (allGroups)
             q->getEvents();
         else
-            q->getEvents(filterGroupIds.toList());
+            q->getEvents(filterGroupIds.values());
     }
 }
 
@@ -93,7 +93,7 @@ void ConversationModelPrivate::groupsDeletedSlot(const QList<int> &groupIds)
         if (allGroups)
             q->getEvents();
         else
-            q->getEvents(filterGroupIds.toList());
+            q->getEvents(filterGroupIds.values());
     }
 }
 
@@ -126,7 +126,7 @@ bool ConversationModelPrivate::acceptsEvent(const Event &event) const
 
 QSqlQuery ConversationModelPrivate::buildQuery() const
 {
-    QList<int> groups = filterGroupIds.toList();
+    QList<int> groups = filterGroupIds.values();
     QString q;
     int unionCount = 0;
 
@@ -247,7 +247,7 @@ bool ConversationModel::setFilter(Event::EventType type,
     d->filterDirection = direction;
 
     if (!d->filterGroupIds.isEmpty()) {
-        return getEvents(d->filterGroupIds.toList());
+        return getEvents(d->filterGroupIds.values());
     } else if (d->allGroups) {
         return getEvents();
     }
@@ -264,7 +264,7 @@ bool ConversationModel::getEvents(QList<int> groupIds)
 {
     Q_D(ConversationModel);
 
-    d->filterGroupIds = QSet<int>::fromList(groupIds);
+    d->filterGroupIds = QSet<int>(groupIds.begin(), groupIds.end());
     d->allGroups = false;
 
     beginResetModel();
diff --git a/src/databaseio.cpp b/src/databaseio.cpp
index dca4263..cefff62 100644
--- a/src/databaseio.cpp
+++ b/src/databaseio.cpp
@@ -864,7 +864,7 @@ bool DatabaseIO::modifyEvent(Event &event)
             if (part.id() >= 0) {
                 if (!idList.isEmpty())
                     idList.append(',');
-                idList.append(QString::number(part.id()));
+                idList.append(QString::number(part.id()).toUtf8());
             }
         }
 
@@ -1059,7 +1059,7 @@ bool DatabaseIO::getGroups(const QString &localUid, const QString &remoteUid, QL
         if (!remoteUid.isEmpty())
             q += "Groups.remoteUids = :remoteUid ";
     }
-    q += "GROUP BY Groups.id " + queryOrder;
+    q += QString("GROUP BY Groups.id " + queryOrder).toUtf8();
 
     QSqlQuery query = CommHistoryDatabase::prepare(q.data(), d->connection());
     if (!localUid.isEmpty())
diff --git a/src/draftsmodel.cpp b/src/draftsmodel.cpp
index 9c02d55..aa24b62 100644
--- a/src/draftsmodel.cpp
+++ b/src/draftsmodel.cpp
@@ -55,13 +55,13 @@ DraftsModel::~DraftsModel()
 QList<int> DraftsModel::filterGroups() const
 {
     Q_D(const DraftsModel);
-    return d->filterGroups.toList();
+    return d->filterGroups.values();
 }
 
 void DraftsModel::setFilterGroups(const QList<int> &list)
 {
     Q_D(DraftsModel);
-    QSet<int> groupIds = list.toSet();
+    QSet<int> groupIds(list.begin(), list.end());
     if (groupIds == d->filterGroups)
         return;
 
@@ -89,7 +89,7 @@ bool DraftsModel::getEvents()
 
     // As in ConversationModel, a UNION ALL is used to get better
     // optimization out of sqlite
-    QList<int> groups = d->filterGroups.toList();
+    QList<int> groups(d->filterGroups.begin(), d->filterGroups.end());
     int unionCount = 0;
     QString q;
     do {
diff --git a/src/event.cpp b/src/event.cpp
index 45ccf89..20ca1b6 100644
--- a/src/event.cpp
+++ b/src/event.cpp
@@ -637,17 +637,17 @@ QList<MessagePart> Event::messageParts() const
 
 QStringList Event::toList() const
 {
-    return d->headers.value(MMS_TO_HEADER).split("\x1e", QString::SkipEmptyParts);
+    return d->headers.value(MMS_TO_HEADER).split("\x1e", Qt::SkipEmptyParts);
 }
 
 QStringList Event::ccList() const
 {
-    return d->headers.value(MMS_CC_HEADER).split("\x1e", QString::SkipEmptyParts);
+    return d->headers.value(MMS_CC_HEADER).split("\x1e", Qt::SkipEmptyParts);
 }
 
 QStringList Event::bccList() const
 {
-    return d->headers.value(MMS_BCC_HEADER).split("\x1e", QString::SkipEmptyParts);
+    return d->headers.value(MMS_BCC_HEADER).split("\x1e", Qt::SkipEmptyParts);
 }
 
 Event::EventReadStatus Event::readStatus() const
diff --git a/src/eventmodel_p.cpp b/src/eventmodel_p.cpp
index bf66529..e3893e1 100644
--- a/src/eventmodel_p.cpp
+++ b/src/eventmodel_p.cpp
@@ -360,7 +360,7 @@ void EventModelPrivate::onDemandResolverFinished()
             resolvedRecipients.insert(*it);
     }
 
-    slotContactChanged(resolvedRecipients.toList());
+    slotContactChanged(resolvedRecipients.values());
 }
 
 void EventModelPrivate::modifyInModel(Event &event)
@@ -537,13 +537,13 @@ void EventModelPrivate::recipientsUpdated(const QSet<Recipient> &recipients, boo
 
 void EventModelPrivate::slotContactInfoChanged(const RecipientList &recipients)
 {
-    QSet<Recipient> changed = QSet<Recipient>::fromList(recipients.recipients());
+    QSet<Recipient> changed = QSet<Recipient>(recipients.recipients().begin(), recipients.recipients().end());
     recipientsUpdated(changed);
 }
 
 void EventModelPrivate::slotContactChanged(const RecipientList &recipients)
 {
-    QSet<Recipient> changed = QSet<Recipient>::fromList(recipients.recipients());
+    QSet<Recipient> changed = QSet<Recipient>(recipients.recipients().begin(), recipients.recipients().end());
     recipientsUpdated(changed, true);
 }
 
diff --git a/src/groupmanager.cpp b/src/groupmanager.cpp
index ba36a00..359a401 100644
--- a/src/groupmanager.cpp
+++ b/src/groupmanager.cpp
@@ -380,7 +380,7 @@ void GroupManagerPrivate::slotContactInfoChanged(const RecipientList &recipients
 {
     Q_Q(GroupManager);
 
-    QSet<Recipient> changed = QSet<Recipient>::fromList(recipients.recipients());
+    QSet<Recipient> changed = QSet<Recipient>(recipients.recipients().begin(), recipients.recipients().end());
 
     foreach (GroupObject *group, groups) {
         if (group->recipients().intersects(changed))
@@ -392,7 +392,7 @@ void GroupManagerPrivate::slotContactChanged(const RecipientList &recipients)
 {
     Q_Q(GroupManager);
 
-    QSet<Recipient> changed = QSet<Recipient>::fromList(recipients.recipients());
+    QSet<Recipient> changed = QSet<Recipient>(recipients.recipients().begin(), recipients.recipients().end());
 
     foreach (GroupObject *group, groups) {
         if (group->recipients().intersects(changed))
diff --git a/src/groupmodel.cpp b/src/groupmodel.cpp
index 978e207..953f6a0 100644
--- a/src/groupmodel.cpp
+++ b/src/groupmodel.cpp
@@ -94,7 +94,7 @@ void GroupModelPrivate::setManager(GroupManager *m)
         connect(manager, SIGNAL(groupsCommitted(QList<int>,bool)), q, SIGNAL(groupsCommitted(QList<int>,bool)));
 
         groups = manager->groups();
-        qSort(groups.begin(), groups.end(), groupObjectSort);
+        std::sort(groups.begin(), groups.end(), groupObjectSort);
     }
 
     q->endResetModel();
diff --git a/src/recentcontactsmodel.cpp b/src/recentcontactsmodel.cpp
index dcb8d6d..459cc69 100644
--- a/src/recentcontactsmodel.cpp
+++ b/src/recentcontactsmodel.cpp
@@ -271,8 +271,8 @@ void RecentContactsModelPrivate::prependEvents(QList<Event> events, bool resolve
         }
 
         // Remove the rows that have been made obsolete
-        QList<int> removeIndices = removeSet.toList();
-        qSort(removeIndices);
+        QList<int> removeIndices = removeSet.values();
+        std::sort(removeIndices.begin(), removeIndices.end());
 
         int count;
         while ((count = removeIndices.count()) != 0) {
diff --git a/src/recipient.cpp b/src/recipient.cpp
index 0a73a5c..ed2f505 100644
--- a/src/recipient.cpp
+++ b/src/recipient.cpp
@@ -457,7 +457,7 @@ QList<int> RecipientList::contactIds() const
         if (r.contactId() > 0)
             re.insert(r.contactId());
     }
-    return re.toList();
+    return re.values();
 }
 
 QStringList RecipientList::remoteUids() const
diff --git a/src/recipienteventmodel.cpp b/src/recipienteventmodel.cpp
index 39a09d0..073608f 100644
--- a/src/recipienteventmodel.cpp
+++ b/src/recipienteventmodel.cpp
@@ -102,8 +102,6 @@ void RecipientEventModelPrivate::fetchEvents()
 
 void RecipientEventModelPrivate::fetcherFinished()
 {
-    Q_Q(RecipientEventModel);
-
     // Our contact is now loaded in the cache
     if (m_contactId > 0) {
         m_recipients = RecipientList::fromContact(m_contactId);
@@ -151,8 +149,6 @@ RecipientEventModel::~RecipientEventModel()
 
 void RecipientEventModel::setRecipients(const Recipient &recipient)
 {
-    Q_D(RecipientEventModel);
-
     setRecipients(RecipientList() << recipient);
 }
 
diff --git a/tests/common.cpp b/tests/common.cpp
index 461f624..bc0b40b 100644
--- a/tests/common.cpp
+++ b/tests/common.cpp
@@ -282,7 +282,7 @@ QList<int> addTestContacts(const QList<QPair<QString, QPair<QString, QString> >
 
 bool addTestContactAddress(int contactId, const QString &remoteUid, const QString &localUid)
 {
-    QContact existingAggregate = manager()->contact(apiContactId(contactId, manager()->managerUri()));
+    QContact existingAggregate = manager()->contact(apiContactId(contactId));
     if (internalContactId(existingAggregate.id()) != (unsigned)contactId) {
         qWarning() << "Could not retrieve contact:" << contactId;
         return false;
@@ -331,7 +331,7 @@ void modifyTestContact(int id, const QString &name, bool favorite)
 {
     qDebug() << Q_FUNC_INFO << id << name;
 
-    QContact existingAggregate = manager()->contact(apiContactId(id, manager()->managerUri()));
+    QContact existingAggregate = manager()->contact(apiContactId(id));
     if (internalContactId(existingAggregate.id()) != (unsigned)id) {
         qWarning() << "Could not retrieve contact:" << id;
         return;
@@ -365,7 +365,7 @@ void modifyTestContact(int id, const QString &name, bool favorite)
 
 void deleteTestContact(int id)
 {
-    const QContactId contactId = apiContactId(id, manager()->managerUri());
+    const QContactId contactId = apiContactId(id);
     const QContact contact = manager()->contact(contactId);
 
     const QContactId localId = localContactForAggregate(contactId);
@@ -507,9 +507,10 @@ void deleteAll()
 QString randomMessage(int words)
 {
     QString msg;
+    QRandomGenerator qrand;
     QTextStream msgStream(&msg, QIODevice::WriteOnly);
     for(int j = 0; j < words; j++) {
-        msgStream << msgWords[qrand() % numWords] << " ";
+        msgStream << msgWords[qrand.generate() % numWords] << " ";
     }
     return msg;
 }
@@ -519,7 +520,7 @@ bool waitSignal(QSignalSpy &spy, int msec)
     if (!spy.isEmpty()) {
         return true;
     }
-    QTime timer;
+    QElapsedTimer timer;
     timer.start();
     while (timer.elapsed() < msec && spy.isEmpty()) {
         QCoreApplication::sendPostedEvents();
@@ -539,7 +540,7 @@ void summarizeResults(const QString &className, QList<int> &times, QFile *logFil
         timeList.append(QString::number(time));
     }
 
-    qSort(times);
+    std::sort(times.begin(), times.end());
     float median = 0.0;
     const int iterations(times.count());
     if (iterations % 2 > 0) {
diff --git a/tests/common.h b/tests/common.h
index 40c92a9..e0887d9 100644
--- a/tests/common.h
+++ b/tests/common.h
@@ -29,6 +29,7 @@
 #include "group.h"
 
 #include <QFile>
+#include <QRandomGenerator>
 #include <QSignalSpy>
 #include <QStandardPaths>
 
@@ -40,9 +41,10 @@ template <class RandomAccessIterator>
 void random_shuffle (RandomAccessIterator first, RandomAccessIterator last)
 {
     typename std::iterator_traits<RandomAccessIterator>::difference_type i, n;
+    QRandomGenerator qrand;
     n = (last-first);
     for (i=n-1; i>0; --i) {
-        qSwap(first[i],first[qrand() % (i+1)]);
+        qSwap(first[i],first[qrand.generate() % (i+1)]);
     }
 }
 
diff --git a/tests/mem_eventmodel/mem_eventmodel.cpp b/tests/mem_eventmodel/mem_eventmodel.cpp
index ec53979..00f2904 100644
--- a/tests/mem_eventmodel/mem_eventmodel.cpp
+++ b/tests/mem_eventmodel/mem_eventmodel.cpp
@@ -37,11 +37,11 @@ Group group;
 #define WAIT_TIMEOUT 30000
 #define CALM_TIMEOUT 500
 
-#define MALLINFO_DUMP(s) {struct mallinfo m = mallinfo();qDebug() << "MALLINFO" << (s) << m.arena << m.uordblks << m.fordblks;}
+#define MALLINFO_DUMP(s) {struct mallinfo2 m = mallinfo2();qDebug() << "MALLINFO" << (s) << m.arena << m.uordblks << m.fordblks;}
 
 static void waitWithDeletes(int msec)
 {
-    QTime timer;
+    QElapsedTimer timer;
     timer.start();
     while (timer.elapsed() < msec) {
         QCoreApplication::sendPostedEvents();
@@ -126,7 +126,7 @@ void MemEventModelTest::addEvents()
 
         waitWithDeletes(100);
         MALLINFO_DUMP("query done");
-        struct mallinfo m = mallinfo();
+        struct mallinfo2 m = mallinfo2();
 
         if (i >= 50)
             mem += m.uordblks - lastMem;
diff --git a/tests/perf_callmodel/callmodelperftest.cpp b/tests/perf_callmodel/callmodelperftest.cpp
index 6959714..c838a3c 100644
--- a/tests/perf_callmodel/callmodelperftest.cpp
+++ b/tests/perf_callmodel/callmodelperftest.cpp
@@ -40,7 +40,7 @@ void CallModelPerfTest::initTestCase()
         logFile = 0;
     }
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void CallModelPerfTest::init()
@@ -103,6 +103,7 @@ void CallModelPerfTest::getEvents()
     qDebug() << Q_FUNC_INFO << "- Creating" << contacts << "contacts";
 
     QList<QPair<QString, QPair<QString, QString> > > contactDetails;
+    QRandomGenerator qrand;
 
     int ci = remoteUids.count();
     while(ci < contacts) {
@@ -201,7 +202,7 @@ void CallModelPerfTest::getEvents()
         fetchModel.setResolveContacts(resolve ? EventModel::ResolveImmediately : EventModel::DoNotResolve);
         fetchModel.setFilter(CallModel::SortByContact);
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
 
         bool result = fetchModel.getEvents();
diff --git a/tests/perf_conversationmodel/conversationmodelperftest.cpp b/tests/perf_conversationmodel/conversationmodelperftest.cpp
index 8d651df..2650938 100644
--- a/tests/perf_conversationmodel/conversationmodelperftest.cpp
+++ b/tests/perf_conversationmodel/conversationmodelperftest.cpp
@@ -41,7 +41,7 @@ void ConversationModelPerfTest::initTestCase()
         logFile = 0;
     }
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void ConversationModelPerfTest::init()
@@ -90,6 +90,7 @@ void ConversationModelPerfTest::getEvents()
     qRegisterMetaType<QModelIndex>("QModelIndex");
 
     QDateTime startTime = QDateTime::currentDateTime();
+    QRandomGenerator qrand;
 
     cleanupTestGroups();
     cleanupTestEvents();
@@ -228,7 +229,7 @@ void ConversationModelPerfTest::getEvents()
             fetchModel.setChunkSize(limit);
         }
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
         bool result = fetchModel.getEvents(groupIds);
         QVERIFY(result);
diff --git a/tests/perf_groupmodel/groupmodelperftest.cpp b/tests/perf_groupmodel/groupmodelperftest.cpp
index bd34a84..f9f64c8 100644
--- a/tests/perf_groupmodel/groupmodelperftest.cpp
+++ b/tests/perf_groupmodel/groupmodelperftest.cpp
@@ -41,7 +41,7 @@ void GroupModelPerfTest::initTestCase()
         logFile = 0;
     }
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void GroupModelPerfTest::init()
@@ -102,6 +102,7 @@ void GroupModelPerfTest::getGroups()
 
     GroupModel addModel;
     QDateTime when = QDateTime::currentDateTime();
+    QRandomGenerator qrand;
 
     qDebug() << Q_FUNC_INFO << "- Creating" << contacts << "contacts";
 
@@ -214,7 +215,7 @@ void GroupModelPerfTest::getGroups()
 
         fetchModel.setManager(&manager);
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
         bool result = fetchModel.getGroups();
         QVERIFY(result);
diff --git a/tests/perf_recentcontactsmodel/recentcontactsmodelperftest.cpp b/tests/perf_recentcontactsmodel/recentcontactsmodelperftest.cpp
index a8a6ade..097b19e 100644
--- a/tests/perf_recentcontactsmodel/recentcontactsmodelperftest.cpp
+++ b/tests/perf_recentcontactsmodel/recentcontactsmodelperftest.cpp
@@ -41,7 +41,7 @@ void RecentContactsModelPerfTest::initTestCase()
         logFile = 0;
     }
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void RecentContactsModelPerfTest::init()
@@ -87,6 +87,7 @@ void RecentContactsModelPerfTest::getEvents()
 
     EventModel addModel;
     QDateTime when = QDateTime::currentDateTime();
+    QRandomGenerator qrand;
 
     qDebug() << Q_FUNC_INFO << "- Creating" << contacts << "contacts";
 
@@ -203,7 +204,7 @@ void RecentContactsModelPerfTest::getEvents()
         if (limit)
             fetchModel.setLimit(limit);
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
 
         bool result = fetchModel.getEvents();
diff --git a/tests/profile_callmodel/callmodelprofiletest.cpp b/tests/profile_callmodel/callmodelprofiletest.cpp
index 4f005f5..aa00609 100644
--- a/tests/profile_callmodel/callmodelprofiletest.cpp
+++ b/tests/profile_callmodel/callmodelprofiletest.cpp
@@ -38,7 +38,7 @@ void CallModelProfileTest::initTestCase()
 
     logFile = 0;
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void CallModelProfileTest::init()
@@ -61,6 +61,7 @@ void CallModelProfileTest::prepare()
     qDebug() << Q_FUNC_INFO << "- Creating" << contacts << "new contacts";
 
     QList<QPair<QString, QPair<QString, QString> > > contactDetails;
+    QRandomGenerator qrand;
 
     int ci = remoteUids.count();
     while(ci < contacts) {
@@ -161,7 +162,7 @@ void CallModelProfileTest::execute()
         fetchModel.setResolveContacts(resolve ? EventModel::ResolveImmediately : EventModel::DoNotResolve);
         fetchModel.setFilter(CallModel::SortByContact);
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
 
         bool result = fetchModel.getEvents();
diff --git a/tests/profile_conversationmodel/conversationmodelprofiletest.cpp b/tests/profile_conversationmodel/conversationmodelprofiletest.cpp
index 6330ede..00ec5b6 100644
--- a/tests/profile_conversationmodel/conversationmodelprofiletest.cpp
+++ b/tests/profile_conversationmodel/conversationmodelprofiletest.cpp
@@ -39,7 +39,7 @@ void ConversationModelProfileTest::initTestCase()
 
     logFile = 0;
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void ConversationModelProfileTest::init()
@@ -58,6 +58,7 @@ void ConversationModelProfileTest::prepare()
     qDebug() << Q_FUNC_INFO << "- Creating" << contacts << "contacts";
 
     QList<QPair<QString, QPair<QString, QString> > > contactDetails;
+    QRandomGenerator qrand;
 
     int ci = remoteUids.count();
     while(ci < contacts) {
@@ -184,7 +185,7 @@ void ConversationModelProfileTest::execute()
             fetchModel.setChunkSize(limit);
         }
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
         bool result = fetchModel.getEvents();
         QVERIFY(result);
diff --git a/tests/profile_groupmodel/groupmodelprofiletest.cpp b/tests/profile_groupmodel/groupmodelprofiletest.cpp
index 93fb90e..3e2b1d7 100644
--- a/tests/profile_groupmodel/groupmodelprofiletest.cpp
+++ b/tests/profile_groupmodel/groupmodelprofiletest.cpp
@@ -37,7 +37,7 @@ void GroupModelProfileTest::initTestCase()
 
     logFile = 0;
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void GroupModelProfileTest::init()
@@ -57,6 +57,7 @@ void GroupModelProfileTest::prepare()
     qDebug() << Q_FUNC_INFO << "- Creating" << contacts << "contacts";
 
     QList<QPair<QString, QPair<QString, QString> > > contactDetails;
+    QRandomGenerator qrand;
 
     int ci = remoteUids.count();
     while(ci < contacts) {
@@ -169,7 +170,7 @@ void GroupModelProfileTest::execute()
 
         fetchModel.setManager(&manager);
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
 
         bool result = fetchModel.getGroups();
diff --git a/tests/profile_recentcontactsmodel/recentcontactsmodelprofiletest.cpp b/tests/profile_recentcontactsmodel/recentcontactsmodelprofiletest.cpp
index 9c8da0f..1f44ce8 100644
--- a/tests/profile_recentcontactsmodel/recentcontactsmodelprofiletest.cpp
+++ b/tests/profile_recentcontactsmodel/recentcontactsmodelprofiletest.cpp
@@ -37,7 +37,7 @@ void RecentContactsModelProfileTest::initTestCase()
 
     logFile = 0;
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator( QDateTime::currentDateTime().toTime_t() );
 }
 
 void RecentContactsModelProfileTest::init()
@@ -60,6 +60,7 @@ void RecentContactsModelProfileTest::prepare()
     qDebug() << Q_FUNC_INFO << "- Creating" << contacts << "contacts";
 
     QList<QPair<QString, QPair<QString, QString> > > contactDetails;
+    QRandomGenerator qrand;
 
     int ci = remoteUids.count();
     while(ci < contacts) {
@@ -173,7 +174,7 @@ void RecentContactsModelProfileTest::execute()
         RecentContactsModel fetchModel;
         fetchModel.setLimit(limit);
 
-        QTime time;
+        QElapsedTimer time;
         time.start();
 
         bool result = fetchModel.getEvents();
diff --git a/tests/tests.xml.in b/tests/tests.xml.in
index c14688b..bf679ed 100644
--- a/tests/tests.xml.in
+++ b/tests/tests.xml.in
@@ -5,9 +5,6 @@
 
        <set name="libcommhistory-qt5-tests-auto" feature="libcommhistory-qt5">
            <description>libcommhistory-qt5 auto tests</description>
-           <case name="mem_eventmodel">
-               <step>@RUN_TEST@ auto mem_eventmodel</step>
-           </case>
            <case name="ut_callmodel" level="Component" type="Functional">
                <step>@RUN_TEST@ auto ut_callmodel</step>
            </case>
diff --git a/tests/unit_tests.pro b/tests/unit_tests.pro
index 7025ca4..e1a7440 100644
--- a/tests/unit_tests.pro
+++ b/tests/unit_tests.pro
@@ -25,7 +25,6 @@
 TEMPLATE = subdirs
 
 SUBDIRS = \
-    mem_eventmodel \
     ut_eventmodel \
     ut_callmodel \
     ut_conversationmodel \
diff --git a/tests/ut_callmodel/callmodeltest.cpp b/tests/ut_callmodel/callmodeltest.cpp
index bf35336..b83168f 100644
--- a/tests/ut_callmodel/callmodeltest.cpp
+++ b/tests/ut_callmodel/callmodeltest.cpp
@@ -65,7 +65,7 @@ void CallModelTest::initTestCase()
 
     QTest::qWait(100);
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator qrand( QDateTime::currentDateTime().toTime_t() );
 
     addTestGroups( group1, group2 );
 
diff --git a/tests/ut_conversationmodel/conversationmodeltest.cpp b/tests/ut_conversationmodel/conversationmodeltest.cpp
index 56cfc62..1ba5c38 100644
--- a/tests/ut_conversationmodel/conversationmodeltest.cpp
+++ b/tests/ut_conversationmodel/conversationmodeltest.cpp
@@ -45,7 +45,7 @@ void ConversationModelTest::initTestCase()
 
     loop = new QEventLoop(this);
 
-    qsrand(QDateTime::currentDateTime().toTime_t());
+    QRandomGenerator qrand(QDateTime::currentDateTime().toTime_t());
 
     addTestGroups(group1, group2);
 
diff --git a/tests/ut_eventmodel/eventmodeltest.cpp b/tests/ut_eventmodel/eventmodeltest.cpp
index 76a107f..20924c6 100644
--- a/tests/ut_eventmodel/eventmodeltest.cpp
+++ b/tests/ut_eventmodel/eventmodeltest.cpp
@@ -62,7 +62,7 @@ void EventModelTest::initTestCase()
 {
     initTestDatabase();
 
-    qsrand(QDateTime::currentDateTime().toTime_t());
+    QRandomGenerator qrand(QDateTime::currentDateTime().toTime_t());
 
     new Adaptor(this);
     QVERIFY(QDBusConnection::sessionBus().registerObject(
@@ -1000,9 +1000,9 @@ void EventModelTest::testCcBcc()
     Event e;
     QVERIFY(model.databaseIO().getEvent(event.id(), e));
     QVERIFY(compareEvents(event, e));
-    QCOMPARE(e.ccList().toSet(), ccList.toSet());
-    QCOMPARE(e.bccList().toSet(), bccList.toSet());
-    QCOMPARE(e.toList().toSet(), toList.toSet());
+    QCOMPARE(QSet<QString>(e.ccList().begin(), e.ccList().end()), QSet<QString>(ccList.begin(), ccList.end()));
+    QCOMPARE(QSet<QString>(e.bccList().begin(),e.bccList().end()), QSet<QString>(bccList.begin(), bccList.end()));
+    QCOMPARE(QSet<QString>(e.toList().begin(),e.toList().end()), QSet<QString>(toList.begin(), toList.end()));
 
     event.resetModifiedProperties();
     ccList.clear();
@@ -1023,9 +1023,9 @@ void EventModelTest::testCcBcc()
 
     QVERIFY(model.databaseIO().getEvent(event.id(), e));
     QVERIFY(compareEvents(event, e));
-    QCOMPARE(e.ccList().toSet(), ccList.toSet());
-    QCOMPARE(e.bccList().toSet(), bccList.toSet());
-    QCOMPARE(e.toList().toSet(), toList.toSet());
+    QCOMPARE(QSet<QString>(e.ccList().begin(), e.ccList().end()), QSet<QString>(ccList.begin(), ccList.end()));
+    QCOMPARE(QSet<QString>(e.bccList().begin(),e.bccList().end()), QSet<QString>(bccList.begin(), bccList.end()));
+    QCOMPARE(QSet<QString>(e.toList().begin(),e.toList().end()), QSet<QString>(toList.begin(), toList.end()));
 
     event.resetModifiedProperties();
     ccList.clear();
diff --git a/tests/ut_groupmodel/groupmodeltest.cpp b/tests/ut_groupmodel/groupmodeltest.cpp
index 2df3ea9..2cd7318 100644
--- a/tests/ut_groupmodel/groupmodeltest.cpp
+++ b/tests/ut_groupmodel/groupmodeltest.cpp
@@ -125,7 +125,7 @@ void GroupModelTest::initTestCase()
 
     loop = new QEventLoop(this);
 
-    qsrand(QDateTime::currentDateTime().toTime_t());
+    QRandomGenerator qrand(QDateTime::currentDateTime().toTime_t());
 }
 
 void GroupModelTest::init()
@@ -664,7 +664,7 @@ void GroupModelTest::streamingQuery()
 
     QList<int> idsOrig;
     QList<int> idsStream;
-    QTime timer;
+
     int count = 0;
     int chunkSize = firstChunkSize;
     while (count < total) {
@@ -716,9 +716,9 @@ void GroupModelTest::streamingQuery()
 
     verifyModelIsReady(streamModel, &modelReady);
     QVERIFY(!streamModel.canFetchMore(QModelIndex()));
-    QCOMPARE(idsOrig.toSet().size(), idsOrig.size());
-    QCOMPARE(idsStream.toSet().size(), idsStream.size());
-    QVERIFY(idsOrig.toSet() == idsStream.toSet());
+    QCOMPARE(QSet<int>(idsOrig.begin(), idsOrig.end()).size(), idsOrig.size());
+    QCOMPARE(QSet<int>(idsStream.begin(), idsStream.end()).size(), idsStream.size());
+    QVERIFY(QSet<int>(idsOrig.begin(), idsOrig.end()) == QSet<int>(idsStream.begin(), idsStream.end()));
 
     modelThread.quit();
     modelThread.wait(5000);
@@ -983,6 +983,8 @@ void GroupModelTest::resolveContact()
     QSignalSpy groupDataChanged(&groupModel, SIGNAL(dataChanged(const QModelIndex &, const QModelIndex &)));
     QSignalSpy groupsCommitted(&groupModel, SIGNAL(groupsCommitted(QList<int>,bool)));
 
+    QRandomGenerator qrand;
+
     QString phoneNumber = QString().setNum(qrand() % 10000000);
     QString contactName = QString("Test Contact 123");
     int contactId = addTestContact(contactName, phoneNumber, RING_ACCOUNT);
diff --git a/tests/ut_recentcontactsmodel/recentcontactsmodeltest.cpp b/tests/ut_recentcontactsmodel/recentcontactsmodeltest.cpp
index 623dce0..bb2784c 100644
--- a/tests/ut_recentcontactsmodel/recentcontactsmodeltest.cpp
+++ b/tests/ut_recentcontactsmodel/recentcontactsmodeltest.cpp
@@ -129,7 +129,7 @@ void RecentContactsModelTest::initTestCase()
     initTestDatabase();
     QVERIFY(QDBusConnection::sessionBus().isConnected());
 
-    qsrand(QDateTime::currentDateTime().toTime_t());
+    QRandomGenerator qsrand(QDateTime::currentDateTime().toTime_t());
 
     addTestGroups( group1, group2 );
 
diff --git a/tests/ut_recipienteventmodel/recipienteventmodeltest.cpp b/tests/ut_recipienteventmodel/recipienteventmodeltest.cpp
index a58da1d..52e9605 100644
--- a/tests/ut_recipienteventmodel/recipienteventmodeltest.cpp
+++ b/tests/ut_recipienteventmodel/recipienteventmodeltest.cpp
@@ -36,7 +36,7 @@ void RecipientEventModelTest::initTestCase()
 {
     initTestDatabase();
 
-    qsrand(QDateTime::currentDateTime().toTime_t());
+    QRandomGenerator(QDateTime::currentDateTime().toTime_t());
 }
 
 void RecipientEventModelTest::cleanupTestCase()
diff --git a/tests/ut_singleeventmodel/singleeventmodeltest.cpp b/tests/ut_singleeventmodel/singleeventmodeltest.cpp
index aa95c11..0351cd5 100644
--- a/tests/ut_singleeventmodel/singleeventmodeltest.cpp
+++ b/tests/ut_singleeventmodel/singleeventmodeltest.cpp
@@ -20,7 +20,7 @@ void SingleEventModelTest::initTestCase()
 {
     initTestDatabase();
 
-    qsrand(QDateTime::currentDateTime().toTime_t());
+    QRandomGenerator(QDateTime::currentDateTime().toTime_t());
 
     addTestGroups(group1, group2);
 }
diff --git a/tools/commhistory-tool.cpp b/tools/commhistory-tool.cpp
index cd65376..fb78e11 100644
--- a/tools/commhistory-tool.cpp
+++ b/tools/commhistory-tool.cpp
@@ -268,7 +268,7 @@ int doAdd(const QStringList &arguments, const QVariantMap &options)
     if (options.contains("-out"))
         direction = Event::Outbound;
 
-    qsrand(QDateTime::currentDateTime().toTime_t());
+    QRandomGenerator qrand(QDateTime::currentDateTime().toTime_t());
     bool randomRemote = true;
     if (arguments.count() > 3) {
         remoteUid = arguments.at(3);
@@ -281,7 +281,7 @@ int doAdd(const QStringList &arguments, const QVariantMap &options)
         Event e;
 
         if (direction == Event::UnknownDirection)
-            e.setDirection((Event::EventDirection)((qrand() & 1) + 1));
+            e.setDirection((Event::EventDirection)((qrand.generate() & 1) + 1));
         else
             e.setDirection(direction);
 
@@ -289,20 +289,20 @@ int doAdd(const QStringList &arguments, const QVariantMap &options)
         {
             e.setType(Event::MMSEvent);
             e.setLocalUid(localUid);
-            e.setSubject(mmsSubject[qrand() % numMmsSubjects]);
+            e.setSubject(mmsSubject[qrand.generate() % numMmsSubjects]);
             e.setMessageToken(QUuid::createUuid().toString());
 
-            if(e.direction() == Event::Outbound || qrand() % 2 == 0)
+            if(e.direction() == Event::Outbound || qrand.generate() % 2 == 0)
             {
-                if(qrand() % 2 == 0)
+                if(qrand.generate() % 2 == 0)
                     e.setCcList(QStringList() << "111111" << "222222" << "iam@cc.list.com");
-                if(qrand() % 2 == 0)
+                if(qrand.generate() % 2 == 0)
                     e.setBccList(QStringList() << "33333" << "44444" << "iam@bcc.list.com");
 
                 QList<MessagePart> parts;
 
                 bool smilAdded = false;
-                if (qrand() % 2 == 0)
+                if (qrand.generate() % 2 == 0)
                 {
                     MessagePart part1;
                     part1.setContentType("application/smil");
@@ -313,7 +313,7 @@ int doAdd(const QStringList &arguments, const QVariantMap &options)
                 part2.setContentId("text_slide1");
                 part2.setContentType("text/plain");
                 parts << part2;
-                if (smilAdded || qrand() % 3 == 0)
+                if (smilAdded || qrand.generate() % 3 == 0)
                 {
                     MessagePart part3;
                     part3.setContentId("catphoto");
@@ -343,15 +343,15 @@ int doAdd(const QStringList &arguments, const QVariantMap &options)
         } else {
             e.setIsRead(false);
         }
-        e.setBytesReceived(qrand() % 1024);
+        e.setBytesReceived(qrand.generate() % 1024);
         e.setGroupId(groupId);
         if (randomRemote) {
-            e.setRecipients(Recipient(e.localUid(), remoteUids[qrand() % numRemoteUids]));
+            e.setRecipients(Recipient(e.localUid(), remoteUids[qrand.generate() % numRemoteUids]));
         } else {
             e.setRecipients(Recipient(e.localUid(), remoteUid));
         }
 
-        e.setFreeText(messageText.isNull() ? textContent[qrand() % numTextContents] : messageText);
+        e.setFreeText(messageText.isNull() ? textContent[qrand.generate() % numTextContents] : messageText);
 
         events.append(e);
     }
@@ -372,7 +372,7 @@ int doAddCall( const QStringList &arguments, const QVariantMap &options )
 {
     Q_UNUSED( options )
 
-    qsrand( QDateTime::currentDateTime().toTime_t() );
+    QRandomGenerator qrand(QDateTime::currentDateTime().toTime_t());
 
     int count = 1;
     if (options.contains("-n")) {
@@ -407,7 +407,7 @@ int doAddCall( const QStringList &arguments, const QVariantMap &options )
         }
 
         if (callType == -1)
-            callType = qrand() % 3;
+            callType = qrand.generate() % 3;
 
         Event::EventDirection direction = Event::Inbound;
         bool isMissed = false;
@@ -476,6 +476,8 @@ int doAddClass0(const QStringList &arguments, const QVariantMap &options)
     Q_UNUSED(arguments)
     Q_UNUSED(options)
 
+    QRandomGenerator qrand(QDateTime::currentDateTime().toTime_t());
+
     QStringList sosRemoteUids;
     sosRemoteUids << "112"
                   << "911"
@@ -491,10 +493,10 @@ int doAddClass0(const QStringList &arguments, const QVariantMap &options)
     Event e;
 
     e.setLocalUid(RING_ACCOUNT);
-    e.setRecipients(Recipient(e.localUid(), sosRemoteUids.at(qrand() % sosRemoteUids.count())));
+    e.setRecipients(Recipient(e.localUid(), sosRemoteUids.at(qrand.generate() % sosRemoteUids.count())));
     e.setDirection(Event::Inbound);
     e.setType(Event::ClassZeroSMSEvent);
-    e.setFreeText(sosMessages.at(qrand() % sosMessages.count()));
+    e.setFreeText(sosMessages.at(qrand.generate() % sosMessages.count()));
     QDateTime now = QDateTime::currentDateTime();
     e.setStartTime(now);
     e.setEndTime(now);
@@ -1177,7 +1179,7 @@ int doJsonImport(const QStringList &arguments, const QVariantMap &options)
             return -1;
         }
 
-        dateOffset = endTime.secsTo(QDateTime(QDate::currentDate()));
+        dateOffset = endTime.secsTo(QDate::currentDate().startOfDay());
     }
 
     bool ok = true;
