From d31d0a9788be96110cdb41e57fdef236037bc7d3 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Fri, 31 Jan 2025 15:09:17 +0100
Subject: [PATCH] disable tests, because it doens't compile

---
 libngf-qt.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libngf-qt.pro b/libngf-qt.pro
index 18f5975..a7edaaa 100644
--- a/libngf-qt.pro
+++ b/libngf-qt.pro
@@ -2,7 +2,7 @@ isEmpty(PREFIX) {
     PREFIX = /usr/local
 }
 TEMPLATE = subdirs
-SUBDIRS += src declarative tests feedback
+SUBDIRS += src declarative feedback
 
 declarative.depends = src
 tests.depends = src declarative
-- 
2.47.1

