From be8cffec61fe9ce428dc40fafd42443b368a0191 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Sat, 8 Jul 2023 16:29:30 +0000
Subject: [PATCH 1/4] use lupdate and lrelease from QT_INSTALL_BINS

---
 fingerterm.pro | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/fingerterm.pro b/fingerterm.pro
index df44fcc..39634dd 100644
--- a/fingerterm.pro
+++ b/fingerterm.pro
@@ -91,7 +91,10 @@ desktopfile.files = $${TARGET}.desktop
 TS_FILE = $$OUT_PWD/fingerterm.ts
 EE_QM = $$OUT_PWD/fingerterm_eng_en.qm
 
-ts.commands += lupdate $$PWD -ts $$TS_FILE
+LUPDATE = $$[QT_INSTALL_BINS]/lupdate
+LRELEASE = $$[QT_INSTALL_BINS]/lrelease
+
+ts.commands += $${LUPDATE} $$PWD -ts $$TS_FILE
 ts.CONFIG += no_check_exist
 ts.output = $$TS_FILE
 ts.input = .
@@ -101,7 +104,7 @@ ts_install.path = /usr/share/translations/source
 ts_install.CONFIG += no_check_exist
 
 # should add -markuntranslated "-" when proper translations are in place (or for testing)
-engineering_english.commands += lrelease -idbased $$TS_FILE -qm $$EE_QM
+engineering_english.commands += $${LRELEASE} -idbased $$TS_FILE -qm $$EE_QM
 engineering_english.CONFIG += no_check_exist
 engineering_english.depends = ts
 engineering_english.input = $$TS_FILE
-- 
2.49.0

