From d3b9be89f75c2ebb0b65e3f85c552ddbd9c1a561 Mon Sep 17 00:00:00 2001
From: Jozef Mlich <jmlich83@gmail.com>
Date: Thu, 24 Aug 2023 18:32:07 +0000
Subject: [PATCH] build with qt6

---
 src/lib/lib.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/lib.pro b/src/lib/lib.pro
index 2282c72..21245a5 100644
--- a/src/lib/lib.pro
+++ b/src/lib/lib.pro
@@ -7,7 +7,7 @@ QT = \
     core \
     qml
 
-PKGCONFIG += mlocale5
+PKGCONFIG += mlocale$${QT_MAJOR_VERSION}
 
 INCLUDEPATH *= ../3rdparty
 
-- 
2.41.0

