From 83488de566f3da376e87d39cd0a23e6b49635a71 Mon Sep 17 00:00:00 2001
From: luka177 <lukapanio@gmail.com>
Date: Sat, 27 Feb 2021 13:59:22 +0200
Subject: [PATCH] fix build

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f69dd99..de1f368 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,8 @@ IF(NOT DEFINED LONG_VERSION OR LONG_VERSION STREQUAL "")
 set(LONG_VERSION ${VERSION})
 ENDIF(NOT DEFINED LONG_VERSION OR LONG_VERSION STREQUAL "")
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+
 find_package(Cor REQUIRED)
 find_package(PkgConfig REQUIRED)
 find_package(Qt5Core REQUIRED)
-- 
2.30.1

