From cba75e8269a209c9a50afa996e7f5bcdaa7f25bb Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Thu, 3 Aug 2023 23:05:50 +0200
Subject: [PATCH 2/2] loader.cpp: Fix plugin suffix for Qt6

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 core/loader.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/loader.cpp b/core/loader.cpp
index aae6055..e8b2e1b 100644
--- a/core/loader.cpp
+++ b/core/loader.cpp
@@ -55,7 +55,11 @@ Loader& Loader::instance()
 
 #define PLUGIN_PREFIX_ENV "SENSORFW_LIBRARY_PATH"
 #define PLUGIN_PREFIX     "lib"
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+#define PLUGIN_SUFFIX     "-qt6.so"
+#else
 #define PLUGIN_SUFFIX     "-qt5.so"
+#endif
 #define SENSOR_SUFFIX     "sensor"
 
 static QString getPluginDirectory()
-- 
2.41.0

